test/e2e/e2e_test.go

```diff
@@ -163,6 +163,68 @@ var _ = Describe("controller", Ordered, func() {
 			})
 		}
 	})
+
+	Context("Azure Templates", func() {
+		var (
+			kc          *kubeclient.KubeClient
+			deleteFunc  func() error
+			clusterName string
+			err         error
+		)
+
+		BeforeAll(func() {
+			By("ensuring Azure credentials are set")
+			kc, err = kubeclient.NewFromLocal(namespace)
+			ExpectWithOffset(2, err).NotTo(HaveOccurred())
+
+			//time.Sleep(60 * time.Second)
+			ExpectWithOffset(2, kc.CreateAzureCredentialsKubeSecret(context.Background())).To(Succeed())
+		})
+
+		AfterEach(func() {
+			// If we failed collect logs from each of the affiliated controllers
+			// as well as the output of clusterctl to store as artifacts.
+			if CurrentSpecReport().Failed() {
+				By("collecting failure logs from controllers")
+				collectLogArtifacts(kc, clusterName, managedcluster.ProviderAWS, managedcluster.ProviderCAPI)
+			}
+
+			// Delete the deployments if they were created.
+			if deleteFunc != nil {
+				By("deleting the deployment")
+				err = deleteFunc()
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+		})
+
+		for _, template := range []managedcluster.Template{
+			managedcluster.TemplateAzureStandaloneCP,
+			managedcluster.TemplateAzureHostedCP,
```

Comment:
You're creating hosted cluster template in the local kind cluster. It should fail. And if it's not - the test should be reworked